src/js/containers/bulkDownload/BulkDownloadPageContainer.jsx

```diff
@@ -180,79 +212,34 @@ export class BulkDownloadPageContainer extends React.Component {
             params.filters.period = formState.period;
         }
 
-        this.requestDownload(params, 'accounts');
+        requestDownload(params, 'accounts');
 
-        logAccountDownload(this.props.bulkDownload.accounts);
-    }
+        logAccountDownload(props.bulkDownload.accounts);
+    };
 
-    requestDownload(params, type) {
-        if (this.request) {
-            this.request.cancel();
+    useEffect(() => {
+        if (prevProps?.match.params.type !== props.match.params.type) {
```

Comment:
I did this in my branches as well but I double checked and it's necessary to check if prevProps != newProp.  https://legacy.reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects

Reply:
ah okay i can fix that, thanks for the tip